src/app/integrations/edit-page/step-configure/step-configure.component.ts

```diff
@@ -106,15 +108,23 @@ export class IntegrationsStepConfigureComponent extends FlowPage
         const step = (this.step = <Step>this.currentFlow.getStep(
           this.position,
         ));
+
+        // If no Step exists, redirect to the Select Step view
         if (!step) {
-          this.router.navigate(['step-select', this.position], {
+          this.router.navigate([ 'step-select', this.position ], {
             relativeTo: this.route.parent,
           });
           return;
         }
+
+        // Step exists, get its configuration
         const stepDef = this.stepStore.getStepConfig(step.stepKind);
-        if (!stepDef || step.stepKind === 'mapper') {
+        log.debugc(
+          () => 'stepConfig: ' + JSON.stringify(stepDef),
+        );
+        if (!stepDef || (step.stepKind === 'mapper' || step.stepKind === 'basic-filter' || step.stepKind === 'advanced-filter')) {
```

Comment:
I'd rather use a switch/case as well, started out with just one other check and grew.